src/EditorFeatures/Core/Diagnostics/AbstractPushOrPullDiagnosticsTaggerProvider.SingleDiagnosticKindPullTaggerProvider.cs

```diff
@@ -117,15 +117,10 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView? textVi
                 // loaded.  This prevents the user from seeing spurious diagnostics while the project is in the process
                 // of loading.  We do keep compiler-syntax as that's based purely on the parse tree, and doesn't need
                 // correct project info to get reasonable results.
-                if (_diagnosticKind != DiagnosticKind.CompilerSyntax)
+                if (_diagnosticKind != DiagnosticKind.CompilerSyntax &&
+                    !await project.IsProjectReadyForSemanticDiagnosticRequestsAsync(cancellationToken).ConfigureAwait(false))
                 {
-                    var service = project.Solution.Services.GetRequiredService<IWorkspaceStatusService>();
-                    if (!await service.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false))
-                        return;
-
-                    using var _ = PooledHashSet<Project>.GetInstance(out var seenProjects);
-                    if (!HasSuccessfullyLoaded(document.Project, seenProjects))
-                        return;
```

Comment:
moved to common location so we can use it from taggers and from LSP.

Reply:
excellent :)

Reply:
though... if that's the case, we have a problem, because i've observed with LSP diagnostics seeing these sorts of errors (see the linked bug).  So i don't get why LSP has a problem, while normal taggers do not.

Reply:
@CyrusNajmabadi I am not sure if the fix (https://github.com/dotnet/roslyn/pull/66569) has flown from Rosyln's release/dev17.5 branch into the main branch and subsequently inserted into the VS repo and into the 17.6 dogfood builds.

Reply:
Seems the fix is not in Roslyn's main branch: https://github.com/dotnet/roslyn/blob/05243d8da8f5f4892dee4125cf300402e14bfc12/src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs#L707-L708

Reply:
Ok.  Whose responsibility is it to make that flow?  Is it yours?  Or infra?

Reply:
The infrastructure team should do it -- I see https://github.com/dotnet/roslyn/pull/66749 is available to carry it although there's a merge conflict so it hasn't happened yet.

Reply:
I resolved that conflict.  Hopefully it can go in asap.